app/models/policy_version.rb

```diff
@@ -86,6 +86,58 @@ def before_save
     self.policy_sha256 = Digest::SHA256.hexdigest(policy_text)
   end
 
+  def after_save
+    log_versions_off_limit
+    delete_versions_off_limit
+  rescue => e
+    Rails.logger.error("Error while ensuring policy '#{id}' versions retention limit: #{e.inspect}")
+  end
+
+  def log_versions_off_limit
+    versions_off_limit.all.each do |policy_version|
+      Rails.logger.debug(
+        "Deleting policy version: #{policy_version.slice(
+          :version,
+          :resource_id,
+          :role_id,
+          :created_at,
+          :client_ip
+        )}]"
+      )
+    end
+  end
+
+  def versions_off_limit
```

Comment:
What do you think about changing this name to `current_versions`?  It looks like it returns the last 20 versions in the `policy_versions` table.  `versions_off` doesn't immediately give me much insight into what it might return.

Reply:
latest_version?